src/openassetio-core/src/hostApi/Manager.cpp

```diff
@@ -410,21 +410,6 @@ void Manager::register_(const EntityReferences &entityReferences,
   if (entityReferences.size() != entityTraitsDatas.size()) {
     throw std::out_of_range{"Parameter lists must be of the same length"};
   }
-
-  if (!entityTraitsDatas.empty()) {
-    const trait::TraitSet firstTraitSet = entityTraitsDatas[0]->traitSet();
-    for (std::size_t idx = 1; idx < entityTraitsDatas.size(); ++idx) {
-      const trait::TraitSet currentTraitSet = entityTraitsDatas[idx]->traitSet();
-      if (currentTraitSet != firstTraitSet) {
-        Str msg = "Mismatched traits at index ";
-        msg += std::to_string(idx);
-        // TODO(DF): expand on error message to include actual trait
-        //  set.
-        throw std::invalid_argument{msg};
-      }
-    }
-  }
```

Comment:
[fix] Don't forget we need to add something so

> A manager can determine if the TraitsData instances supplied to the register method share a common trait set with a single line of code.

Reply:
Ah whoops, yup missed this.  I recall that from a PO perspective you weren't too worried what the shape of such a function should be.  But would be good to at least get together with @elliotcmorris to sketch a quick design.

If time is an issue, this seems like the sort of thing that could be a separate PR(?)

Reply:
Capturing offline discussion - bump to a separate PR.

Reply:
Made an issue: #1042